### PR TITLE
Add log_train_loss_on_step toggle to EasySyntax

### PR DESCRIPTION
**Pull request by sevmag**
## Summary
- Adds an opt-in `log_train_loss_on_step` flag to `EasySyntax` that logs a per-step `train_loss_step` metric in addition to the existing epoch-aggregated `train_loss`.
- Default is `False`, so existing logging behavior is unchanged.

addresses #882

### CODE REVIEW

**Comment by christianlocatelli** [comment]
I will look at this.

File: src/graphnet/models/easy_model.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def __init__(
         scheduler_class: Optional[type] = None,
         scheduler_kwargs: Optional[Dict] = None,
         scheduler_config: Optional[Dict] = None,
+        log_train_loss_on_step: bool = False,
```
Comment (christianlocatelli) [inline src/graphnet/models/easy_model.py]: The variable name could be renamed to `also_log_train_loss_per_step`. This would immediately clarify, that it is an additional option for logging the per-batch loss under a different key. 

```suggestion
        also_log_train_loss_per_step: bool = False,
```
It could be also useful to add a Docstring explaining the arguments in `__init__()`, but especially for `also_log_train_loss_per_step`.
```python
    """
    Args:
        also_log_train_loss_per_step:
            If `True`, logs an additional per-batch metric (`train_loss_step`)
            alongside the existing per-epoch metric (`train_loss`). This can
            be useful for debugging training instabilities or monitoring
            convergence within long epochs.
    """
```

File: src/graphnet/models/easy_model.py [hunk 2]
```diff
@@ -52,6 +53,7 @@ def __init__(
         self._scheduler_class = scheduler_class
         self._scheduler_kwargs = scheduler_kwargs or dict()
         self._scheduler_config = scheduler_config or dict()
+        self._log_train_loss_on_step = log_train_loss_on_step
```
Comment (christianlocatelli) [inline src/graphnet/models/easy_model.py]:
```suggestion
        self._also_log_train_loss_per_step = also_log_train_loss_per_step
```

File: src/graphnet/models/easy_model.py [hunk 3]
```diff
@@ -243,15 +245,26 @@ def training_step(
...
         )
+        if self._log_train_loss_on_step:
+            self.log(
+                "train_loss_step",
+                loss,
+                batch_size=batch_size,
+                prog_bar=False,
+                on_epoch=False,
+                on_step=True,
+                sync_dist=True,
```
Comment (christianlocatelli) [inline src/graphnet/models/easy_model.py]: It might be computationally expensive, if `sync_dist=True`.
There would be syncing across GPUs on every batch, which quickly adds up for high batch number.  It should be maybe clarified in the Docstring at the top, that the training might be slowed down. The default of this option could also be set to `sync_dist=False`.

```suggestion
        if self._also_log_train_loss_on_step:
            self.log(
                "train_loss_step",
                loss,
                batch_size=batch_size,
                prog_bar=False,
                on_epoch=False,
                on_step=True,
                sync_dist=True,
```

**Review by christianlocatelli — COMMENTED** [review]
I left some comments about optional naming and doc improvements.

File: src/graphnet/models/easy_model.py (hunk 3 above)
Comment (sevmag) [reply src/graphnet/models/easy_model.py]: Good point! Let's set sync_dist to false as the default

**Comment by sevmag** [comment]
Hey @christianlocatelli, I implemented your suggestions. Sorry for the commit name, that was claude 😅

**Review by christianlocatelli — COMMENTED** [review]
This looks good to me, thanks for editing the code 👍

**Comment by sevmag** [comment]
@Aske-Rosted tagging you here for completeness (and a potential approval 😅 )

File: src/graphnet/models/easy_model.py [hunk 4]
```diff
@@ -36,8 +36,23 @@ def __init__(
         scheduler_class: Optional[type] = None,
         scheduler_kwargs: Optional[Dict] = None,
         scheduler_config: Optional[Dict] = None,
+        also_log_train_loss_per_step: bool = False,
```
Comment (Aske-Rosted) [inline src/graphnet/models/easy_model.py]: I am sorry for being a little bit annoying  here but I think it is better to just expose the `on_epoch` and `on_step` from the lightning library then the user can decide themselves whether they want just on step, or just on epoch, on both or on neither. The standard values should follow current behavior.

```suggestion
        log_on_epoch: bool = True
        log_on_step: bool = False,
```

File: src/graphnet/models/easy_model.py [hunk 5]
```diff
@@ -243,15 +259,26 @@ def training_step(
...
         loss = self.shared_step(train_batch, batch_idx)
+        batch_size = self._get_batch_size(train_batch)
         self.log(
             "train_loss",
             loss,
-            batch_size=self._get_batch_size(train_batch),
+            batch_size=batch_size,
             prog_bar=True,
             on_epoch=True,
             on_step=False,
```
Comment (Aske-Rosted) [inline src/graphnet/models/easy_model.py]: And then expose the logging setting to the class. This removes duplicate code and then we don't have to define the batch_size. 

```suggestion
on_epoch=log_on_epoch,
on_step=log_on_step,
```

Comment (sevmag) [reply src/graphnet/models/easy_model.py]: makes sense, but how do we want to handle the logging of the val loss? `log_on_epoch` and `log_on_step` for me sounds like you log both val and train on epoch and or step, which I think could also be valid (personally I log train on log and step and val only on epoch). As long as we agree on something together I think either way is fine

Comment (Aske-Rosted) [reply src/graphnet/models/easy_model.py]: I think it is fine to have the logging of the validation and train loss behave in the same way. In principle we could separate the arguments for validation and training, but I think that is a little too many arguments, and moving towards instances where people should just create their own torch-lightning callbacks.

**Review by Aske-Rosted — APPROVED** [review]
Looks good to me now.
